File: ChessApp/UI/Items/ChessboardItem.cs
using ChessEngine;
using ChessEngine.Board;

namespace UI.Items
{
    public class ChessboardItem : Java.Lang.Object
    {
        public readonly BoardSquare boardSquare;
        public readonly Player? player;
        public readonly Piece? piece;
        public readonly CellColor color;

        public ChessboardItem(BoardSquare boardSquare, Player? player, Piece? piece, CellColor color)
        {
            this.boardSquare = boardSquare;
            this.player = player;
            this.piece = piece;
            this.color = color;
        }

        public enum CellColor
        {
            Dark,
            Light
        }
    }
}

File: android/ui/src/main/java/ru/maksimbulva/ui/chessboard/items/ChessboardItem.kt
package ru.maksimbulva.ui.chessboard.items

import ru.maksimbulva.chess.core.engine.Piece
import ru.maksimbulva.chess.core.engine.Player
import ru.maksimbulva.chess.core.engine.board.Cell

class ChessboardItem(
    val cell: Cell,
    val player: Player?,
    val piece: Piece?
)
